tests/unit/test_charm.py

```diff
@@ -83,11 +85,50 @@ def test_given_private_key_is_stored_when_certificates_relation_joined_then_cert
         self.harness.add_relation_unit(
             relation_id=relation_id, remote_unit_name="certificates-provider/0"
         )
-
+        unit_number = self.harness.model.unit.name.split("/")[1]
+        patch_generate_csr.assert_called_with(
+            private_key=PRIVATE_KEY.encode(),
+            private_key_password=PRIVATE_KEY_PASSWORD.encode(),
+            subject=f"{self.harness.charm.app.name}-{unit_number}.{self.harness.model.name}",
```

Comment:
Wouldn't it make sense to extract this into a separate test case like you did in `test_given_common_name_config_is_set_when_certificates_relation_joined_then_certificate_is_requested` ? 
But I'm fine keeping it here too.

Reply:
I'd prefer keeping it here as those 2 behaviors aren't really logically separatable, we use the CSR we generated and place it in the certificate request call.